core-tests/src/test/java/org/mule/runtime/core/api/context/DefaultMuleContextFactoryTestCase.java

```diff
@@ -218,6 +235,46 @@ public void notifiesMuleContextEvents() throws InitialisationException, Configur
     inOrder.verify(listener).onInitialization(eq(context), any(Registry.class));
   }
 
+  @Test
+  public void testCreateMuleContextWithCustomServices() throws InitialisationException, ConfigurationException {
+    MinimalConfigurationBuilder minimalConfigurationBuilder = new MinimalConfigurationBuilder();
+
+    Object testCustomServiceImpl = new Object();
+    Object customQueueManagerImpl = new Object();
+    Object customSecurityManagerImpl = new Object();
+    Map<String, Consumer<ServiceInterceptor<Object>>> interceptors = new HashMap<>();
+    // replace the default Security Manager
+    interceptors.put(OBJECT_SECURITY_MANAGER, serviceInterceptor -> {
+      assertThat(serviceInterceptor.getDefaultServiceImpl().isPresent(), is(true));
```

Comment:
isn't it better to assert that the default is actually the expected instance?

Reply:
I don't have access to the default value, but I could subclass `MinimalConfigurationBuilder` to have control over it.